Sources/FoundationEssentials/String/StringBlocks.swift

```diff
@@ -181,3 +181,26 @@ extension BidirectionalCollection where Element == UTF8.CodeUnit {
         return _StringBlock(start: start, end: end, contentsEnd: contentsEnd)
     }
 }
+
+extension Substring.UTF8View {
+    // Note: we could have these defined on BidirectionalCollection<UInt8>, like _getBlock.
+    // However, all current call sites are funneling through Substring.UTF8View, and it makes
+    // sense to avoid dealing with generics unless we're forced to. Feel free to convert these
+    // utilities to generic functions if needed.
+
+    internal func _lineBounds(
+        around range: Range<Index>
+    ) -> (start: Index, end: Index, contentsEnd: Index) {
+        let result = _getBlock(
+            for: [.findStart, .findEnd, .findContentsEnd, .stopAtLineSeparators], in: range)
+        return (result.start!, result.end!, result.contentsEnd!)
```

Comment:
oh yeah, you're right that we can force unwrap these indices here and below since it should match the options passed into `_getBlock`

Reply:
Yep! (I think at some point it would be worth splitting _getBlock up into separate functions (one each for the findStart/findEnd cases) to eliminate these optional results.)